enterprise/server/remote_execution/containers/ociruntime/ociruntime.go

```diff
@@ -1508,3 +1519,23 @@ func withImageConfig(cmd *repb.Command, image *Image) (*repb.Command, error) {
 func tmpSuffix() string {
 	return fmt.Sprintf(".%d.tmp", mrand.Int64N(1e18))
 }
+
+// Removes any versioned image cache directories under the given path which
+// do not match the current version.
+func cleanStaleImageCacheDirs(root string) error {
+	entries, err := os.ReadDir(root)
+	if err != nil {
+		return fmt.Errorf("read dir %q: %w", root, err)
+	}
+	for _, e := range entries {
+		if e.Name() == imageCacheVersion || !versionDirRegexp.MatchString(e.Name()) {
+			continue
+		}
+		path := filepath.Join(root, e.Name())
+		log.Infof("Removing stale image cache at %q", path)
+		if err := os.RemoveAll(path); err != nil {
+			return fmt.Errorf("remove %q: %w", path, err)
+		}
+	}
+	return nil
+}
```

Comment:
nit: I wonder if this is fast enough to run in production on deployment like this, especially when multiple actions could be calling this at the same time(?)

It feels like if we have to roll back this change for w/e reasons, this could be problematic as well.

Reply:
* This is only called once on startup - shouldn't be called concurrently.
* I don't think rollbacks are worth worrying about - the worst that happens is that images have to be re-pulled

Reply:
ah nvm, this is called inside NewProvider